### PR TITLE
Add ability to create filters to change the logging level for some queries

### LINKED ISSUE #9: It should be possible to add log level filters in logging middleware

### PR DESCRIPTION
**Pull request by Soulou**
Fixes #9

### CODE REVIEW

File: logging_middleware.go [hunk 1]
```diff
@@ -3,21 +3,54 @@ package handlers
...
 }
 
+func NewLoggingMiddlewareWithFilters(logger logrus.FieldLogger, filters map[string]logrus.Level) (*LoggingMiddleware, error) {
+	refilters := map[string]patternInfo{}
+	for pattern, level := range filters {
+		re, err := regexp.Compile(pattern)
+		if err != nil {
+			return nil, errgo.Notef(err, "invalid regexp '%v'", pattern)
+		}
+		refilters[pattern] = patternInfo{re: re, level: level}
```
Comment (Soulou) [inline logging_middleware.go]: No possible to use the regexp as key, as a pointer can't be a key in a map. (And a regexp should be a pointer as there is a mutex in it)

File: logging_middleware.go [hunk 2]
```diff
@@ -3,21 +3,54 @@ package handlers
...
+
+type patternInfo struct {
+	re    *regexp.Regexp
+	level logrus.Level
+}
+
 type LoggingMiddleware struct {
-	logger logrus.FieldLogger
+	logger  logrus.FieldLogger
+	filters map[string]patternInfo
```
Comment (johnsudaar) [inline logging_middleware.go]: Why making this a map if you are always using this like an array ?

Comment (Soulou) [reply logging_middleware.go]: True, it doesn't make a lot of sense

Comment (Soulou) [reply logging_middleware.go]: done

**Comment by Soulou** [comment]
Specs added

**Pull request merged**